diff --git a/lalpulsar/bin/Weave/testWeave_interpolating.sh b/lalpulsar/bin/Weave/testWeave_interpolating.sh
index 55bbd9e52d..77c2d7037b 100644
--- a/lalpulsar/bin/Weave/testWeave_interpolating.sh
+++ b/lalpulsar/bin/Weave/testWeave_interpolating.sh
@@ -90,7 +90,7 @@ if lalpulsar_WeaveCompare --setup-file=WeaveSetup.fits --result-file-1=WeaveOut.
     exitcode=0
 else
     exitcode=77
-    lalpulsar_WeaveCompare --setup-file=WeaveSetup.fits --result-file-1=WeaveOut.fits --result-file-2=RefWeaveOut.fits --param-tol-mism=0
+    #lalpulsar_WeaveCompare --setup-file=WeaveSetup.fits --result-file-1=WeaveOut.fits --result-file-2=RefWeaveOut.fits --param-tol-mism=0
 fi
 set +x
 echo
